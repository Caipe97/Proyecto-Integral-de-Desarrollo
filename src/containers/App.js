import React from "react";;

import Login from './Login';
import Register from './Register';
import Home from './Home';
import Profile from './Profile';
import ResetPassword from './ResetPassword';

import { Switch, Route } from "react-router-dom";

function App() {
    return (
      <div>
        <Switch>
          <Route exact path="/" component={Home}/>
          <Route exact path="/login" component={Login} />
          <Route exact path="/resetPassword" component={ResetPassword} />
          <Route exact path="/register" component={Register} />
          <Route exact path="/profile" component={Profile}/>
        </Switch>
      </div>
    );
}

export default App;