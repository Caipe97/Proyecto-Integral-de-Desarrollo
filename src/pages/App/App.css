h1 {
  font-family: 'SEGA LOGO FONT';
  font-weight: 150;
  color: #0ccac4;
}
.contenedor {
  width: 90%;
  max-width: 1000px;
  margin: 20px auto;
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(4, auto);

  grid-template-areas:
    'header header header'
    'contenido contenido sidebar'
    'widget1 widget2 sidebar'
    'footer footer footer';
  /* DEFINIMOS AREAS entonces tenemos 3 columnas y 4 filas. Luego se le asigna
  a cada parte de la pagina web y bueno eso */
}

.contenedor > div,
.contenedor .header,
.contenedor .contenido,
.contenedor .sidebar,
.contenedor .footer {
  background: #fff;
  padding: 20px;
  border-radius: 4px;
}

.contenedor .widget1 {
  grid-area: widget1;
}
.contenedor .widget2 {
  grid-area: widget2;
}
.contenedor .header {
  background: #12203e;
  grid-area: header;
  align-content: space-between;

  /* grid-column-start:1;
  grid-column-end: 4; */
  /* sino podes indicar con end -1 si no sabes las columnas q hiciste en grid */
}

.contenedor .contenido {
  grid-area: contenido;
  /* grid-column:  span 2; */
}

.contenedor .sidebar {
  grid-column: 3/4;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 100px;
  grid-area: sidebar;
  /* grid-row:span 2; */
}

.contenedor .widget1,
.contenedor .widget2 {
  height: 100px;
  text-align: center;
  align-items: center;
  justify-content: center;
}

.contenedor .footer {
  background: #12203e;
  color: #fff;
  grid-area: footer;
  /* grid-column: span 3; */
}
/* cuando baje de 768 la dimension va a cambiar de las componentes */
/* @media (max-width:768px){
  .contenedor{
    background: #000;
  }
  .contenedor .contenido{
    grid-column: 1/-1;
  }
  .contenedor .sidebar{
    grid-column: 1/-1;
  }
  .contenedor .widget1{
    grid-column: span 2;
  }
} */
/* lo comentado es con media query */

@media (max-width: 768px) {
  .contenedor {
    grid-template-areas:
      'header header header'
      'contenido contenido contenido'
      'sidebar sidebar sidebar'
      'widget1 widget1 widget1 '
      'widget2 widget2 widget2 '
      'footer footer footer';
  }
}
/* con grid area tenes que definir cuando baje de 768px Y LUEGO va a funcar */

.box {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: '100%';
  height: '100%';
  /* background: red;
  color: white; */
}

/* Screen Reset */

.contenedorReset {
  width: 90%;
  max-width: 1000px;
  margin: 20px auto;
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, auto);

  grid-template-areas:
    'headerReset headerReset headerReset'
    'contenidoReset contenidoReset contenidoReset'
    'footer footer footer';

  /* DEFINIMOS AREAS entonces tenemos 3 columnas y 4 filas. Luego se le asigna
  a cada parte de la pagina web y bueno eso */
}

.contenedorReset .contenidoReset {
  grid-area: contenidoReset;
  /* grid-column:  span 2; */
}
.contenedorReset .headerReset {
  grid-area: headerReset;
  background: #12203e;
  align-content: space-between;
  padding: 20px;
  border-radius: 4px;
}

.contenedorReset .footerReset {
  background: #fff;
  padding: 20px;
  border-radius: 4px;
  background: #12203e;
  color: #fff;
  grid-area: footer;
}
