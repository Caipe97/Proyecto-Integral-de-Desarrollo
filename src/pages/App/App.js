import React from "react";
import Login from '../Login/Login';
import Register from '../Register/Register';
import Profile from '../Profile/Profile';
import ResetPassword from '../ResetPassword/ResetPassword';
import Meals from '../Meals/Meals';
import Goals from '../Goals/Goals'
import { Switch, Route } from "react-router-dom";


function App() {
    return (
      <div>
        <Switch>
          <Route exact path="/" component={Login}/>
          <Route exact path="/resetPassword" component={ResetPassword} />
          <Route exact path="/register" component={Register} />
          <Route exact path="/profile" component={Profile}/>
          <Route exact path="/meals" component={Meals}/>
          <Route exact path="/goals" component={Goals}/>
        </Switch>
      </div>
    );
}

export default App;