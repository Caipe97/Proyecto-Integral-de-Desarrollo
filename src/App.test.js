import React from 'react';
import { shallow } from 'enzyme'; 
import App from './containers/App';

it('expect to render App component', () => {
    expect(shallow(<App/>)).toMatchSnapshot();
})
