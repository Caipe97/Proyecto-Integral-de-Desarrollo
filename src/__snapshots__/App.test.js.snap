// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render App component 1`] = `
<div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/login"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/resetPassword"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/register"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/profile"
    />
  </Switch>
</div>
`;
